process/block/metablock.go

```diff
@@ -1363,17 +1365,20 @@ func (mp *metaProcessor) updateState(lastMetaBlock data.MetaHeaderHandler) {
 		return
 	}
 
+	lastMetaBlockRootHash := process.GetScheduledRootHash(lastMetaBlockHash, lastMetaBlock.GetRootHash(), mp.store, mp.marshalizer)
+	prevHeaderRootHash := process.GetScheduledRootHash(prevHeaderHash, prevHeader.GetRootHash(), mp.store, mp.marshalizer)
+
 	if lastMetaBlock.IsStartOfEpochBlock() {
-		log.Debug("trie snapshot", "rootHash", lastMetaBlock.GetRootHash())
-		mp.accountsDB[state.UserAccountsState].SnapshotState(lastMetaBlock.GetRootHash())
+		log.Debug("trie snapshot", "rootHash", lastMetaBlock.GetRootHash(), "scheduledRootHash", lastMetaBlockRootHash)
```

Comment:
Yes here is not needed right now, but I just prepared the flow if someday we will have scheduled in meta. I will remove them for now to avoid the confusion.